docs/excel/excel-add-ins-pivots.md

```diff
@@ -0,0 +1,216 @@
+
+	await context.sync();
+});
+```
+
+## Use an existing pivot table
+Manually created pivot tables are also accessible through the worksheet’s pivot table collection. 
+The following code gets the first pivot table in the worksheet. It then gives the table a name for easy future reference.
+
+```ts
+await Excel.run(async (context) => {
+	let pivotTableList = context.workbook.worksheets.getActiveWorksheet().pivotTables;
+	pivotTableList.load("items");
+	await context.sync();
+
+	let pivotTable = pivotTableList.items[0];
+	pivotTable.name = "My Pivot";
+	await context.sync();
+});
+```
+
+## Add rows and columns to a pivot table
+
+Rows and columns pivot the data around those fields’ values.
+
+Adding the **Farm** column pivots all the sales around each farm. Adding the **Type** and **Classification** rows further breaks down the data based on what fruit was sold and whether it was organic or not.
+
+![A pivot table with a Farm column and Type and Classification rows.](../images/excel-pivots-table-rows-and-columns.png)
+
+```ts
+await Excel.run(async (context) => {
+	let pivotTable = context.workbook.worksheets.getActiveWorksheet().pivotTables.getItem("Farm Sales");
+
+	pivotTable.rowHierarchies.add(pivotTable.hierarchies.getItem("Type"));
+	pivotTable.rowHierarchies.add(pivotTable.hierarchies.getItem("Classification"));
+
+	pivotTable.columnHierarchies.add(pivotTable.hierarchies.getItem("Farm"));
+
+	await context.sync();
+});
+```
+
+You can also have a pivot table with only rows or columns.
+
+```ts
+await Excel.run(async (context) => {
+	let pivotTable = context.workbook.worksheets.getActiveWorksheet().pivotTables.getItem("Farm Sales");
+	pivotTable.rowHierarchies.add(pivotTable.hierarchies.getItem("Farm"));
+	pivotTable.rowHierarchies.add(pivotTable.hierarchies.getItem("Type"));
+	pivotTable.rowHierarchies.add(pivotTable.hierarchies.getItem("Classification"));
+	
+	await context.sync();
+});
+```
+
+## Add data hierarchies to the pivot table
+
+Data hierarchies fill the pivot table with information to combine based on the rows and columns. 
+Adding the data hierarchies of **Crates Sold at Farm** and **Crates Sold Wholesale** gives sums of those figures for each row and column. 
+In the example, both **Farm** and **Type** are rows, with the crate sales as the data. 
+
+![A pivot table showing the total sales of different fruit based on the farm they came from.](../images/excel-pivots-data-hierarchy.png)
+
+```ts
+await Excel.run(async (context) => {
+	let pivotTable = context.workbook.worksheets.getActiveWorksheet().pivotTables.getItem("Farm Sales");
+
+	pivotTable.rowHierarchies.add(pivotTable.hierarchies.getItem("Farm"));
+	pivotTable.rowHierarchies.add(pivotTable.hierarchies.getItem("Type"));
+
+	pivotTable.dataHierarchies.add(pivotTable.hierarchies.getItem("Crates Sold at Farm"));
+	pivotTable.dataHierarchies.add(pivotTable.hierarchies.getItem("Crates Sold Wholesale"));
+
+	await context.sync();
+});
+```
+
+## Change aggregation function
```

Comment:
I think this section needs some code. For example, `summarizeBy` is a parameter to _what_ function?

Reply:
OK. Thise seems better.